client/client.go

```diff
@@ -509,6 +509,51 @@ func (r *NotaryRepository) GetChangelist() (changelist.Changelist, error) {
 	return cl, nil
 }
 
+// GetDelegationRoles returns the keys and roles of the repository's delegations
+func (r *NotaryRepository) GetDelegationRoles() ([]*data.Role, error) {
+	// Update state of the repo to latest
+	_, err := r.Update(false)
```

Comment:
Also, for test purposes, if I remember right you can make `Update` fail by setting `r.tufRepo = nil`